### DISCUSSION TITLE
Confirmed working configuration

### DISCUSSION THREAD

**Discussion by xvdessel** [roelderickx/connective-plugin-linux#20] [General]
I can confirm that this setup works:
Fedora 38 (kernel 6.6.6)
eid-mw 5.1.15
pcsc-lite 1.9.9
VASCO Digipass 905 Smart reader
Google chrome 120

Regards,
Xavier

**Comment by roelderickx** [comment]
@jovanbulck confirmed his setup:
* Ubuntu 22
* Alcor Micro AU9540 reader built-in a HP EliteBook 840 14 inch G9 Notebook PC

**Comment by equaeghe** [comment]
**I can confirm that this code works to get itsme activated.**

System details:

- up-to-date stable Gentoo
- Python 3.12.6
- pcsc-lite 2.0.1
- Firefox 131.0
- Smartcard reader: AU9540 built-in to Lenovo T14

I installed the missing dependency nativemessaging-ng in a venv, as it would not be installed through Gentoo's native package manager. I then ran the instructions in that venv. For some reason, I had to install requests there as well, even though I had it installed, but I got a ‘not found’ error.

I got

- Connective signing extension 1.0.5

I used it for activating itsme for a new phone (I guess I used a borrowed a Windows system previously). That worked by insisting, i.e., trying a second time, as the first time, there occurred an error when reading data from my eID.

**Comment by berenddeschouwer** [comment]
Works

- Fedora 41 beta
- Firefox 131.0.3
- eid 5.1.18
- Connective signing extension 1.0.5
- Smartcard reader: Intopic CR-35 "Generic USB2.0-CRW [Smart Card Reader Interface] 00" (usb id 0bda:0169)

First attempt failed, second attempt succeeded.

**Comment by marnux** [comment]
Confirmed working (for what i need it)

- Debian Sid
- Firefox 128.5.0 Extended support release
- eid 5.1.18
- Connective signing extension 1.0.5
- Reader: VASCO DIGIPASS 870 0 card reader


I installed the dependencies in the Debian python3 stack, as there was an issue compiling pyscard, as no headers found, in the virtual env. That allowed the first part of the setup. For the nativemessaging-ng, i then needed to create a virtual env to get that manifest installed.

Thank you very much for this piece of software. It's a life saver.

**Reply by roelderickx** [reply]
It is not super urgent, but when I find the time I'll document the use of the virtual env, which is the preferred way of installing the manifest. I created #31 for follow up.

**Reply by marnux** [reply]
I think i made the full venv setup work on Debian Sid, by installing libpcsclite-dev package, to get the smartcard related header files. That way it seemed to compile the wheel for pyscard correctly.

**Comment by EricLafortune** [comment]
Works flawlessly (without the virtual env) on:

- Linux Mint 21.1
- Firefox 133.0.3
- Connective signing extension 1.0.5
- Reader: Advanced Card Systems, ACR39U

Thanks!

**Comment by lucascool12** [comment]
Works perfectly, thanks!

- NixOS 24.11
- Firefox 135.0.1
- Connective signing extension 1.0.5
- Reader: VASCO digipas 905

<details>

<summary> NixOS specific things </summary>

Instead of providing a global python interpreter, I opted for creating a derivation with the needed dependencies for the `connective-backend.py` file. And then I changed the path in the `~/.mozilla/native-messaging-hosts/com.connective.signer.json` to the result of the build of the provided derivation (`nix-build connective-plugin-linux.nix`). I hope this helps any future NixOS users.

<details>

<summary> connective-backend.py derivation </summary>

```nix
let
  pkgs = import <nixpkgs> {};
  connective-plugin-linux = {
    stdenv,
    fetchFromGitHub,
    python3,
  }:
  stdenv.mkDerivation {
    name = "connective-plugin-linux";
    version = "2025-03-06";
    pyproject = false;

    src = fetchFromGitHub {
      owner = "roelderickx";
      repo = "connective-plugin-linux";
      rev = "e5438e66b4161419d1e27bbe6bf636c397304de2";
      hash = "sha256-wtclPqccLh/UE5tdloO48AtKgaTi8zNN4NIIqpN+6/U=";
    };

    propagatedBuildInputs = [
      (python3.withPackages (pythonPackages: with pythonPackages; [
        tkinter
        pyscard
        requests
      ]))
    ];

    installPhase = ''
      mkdir -p $out/bin
      install -Dm755 connective-backend.py $out/bin/
    '';
  };
in pkgs.callPackage connective-plugin-linux {}
```
</details>

</details>

**Comment by kirmaha** [comment]
Hurray, I could run your beautiful script:

- Debian Stable 13 (trixie) x86_64
- Firefox ESR 140.5.0
- Connective signing extension 1.0.5
- Reader: ACS ACR38U-CFC-BLI
- Usecase: reactivate Itsme (on a new phone) with a Belgian ID

```bash
# Needed during install process (after `sudo apt-get install libpcsclite-dev python3-pyscard python3-tk`):
sudo apt install python3.13-venv

# Create and enter venv:
python3 -m venv venv-nm (defaults in ~/)
source venv-nm/bin/activate

# Download the code repo as .zip, then unzip, then cd to the unzipped repository and install requirements
cd ~/Downloads/connective-plugin-linux-main
python3 -m pip install requests
pip install -r optional-requirements.txt

# Run script, then link it to the backend
python get_connective_plugin.py
nativemessaging-ng install firefox
```

**Comment by installe** [comment]
I confirm that this works for Debian GNU/Linux 12.12.
I did have to install the following additional packages:

python3-tk
python3-pyscard

I expressed my anger at itsme support for not supporting Linux, I've told them about your repository, told them they ought to compensate you for your work, and build a tutorial or debian package around it.

**Reply by roelderickx** [reply]
> I expressed my anger at itsme support for not supporting Linux, I've told them about your repository, told them they ought to compensate you for your work, and build a tutorial or debian package around it.

It is not an Itsme issue but instead a Connective limitation. They decided Linux support is unnecessary but in fact they should provide support, not me. And maybe they should pay me, but to be honest this project came to exist out of frustration, not greed.

That being said, Itsme is defective in the same way since they are limited to smartphones. Apparently we all agreed a private company is allowed to store our identities, but then at least I'd expect something useful in return. Being able to log in or sign documents using a PC for example. It shouldn't be too hard to implement, because ironically that's exactly what happens when you sign up for Itsme. If you look at it that way it's unclear what problem they were trying to solve in the first place.

Then it is rather inappropriate how they brag about [their](https://www.vrt.be/vrtnws/nl/2019/06/06/de-app-itsme-heeft-in-twee-jaar-een-miljoen-gebruikers/) [significant](https://www.vrt.be/vrtmax/luister/radio/fragmenten/fragment~ffb1e2db-5322-49d9-8faa-e4b2b8c3a28c/) [user](https://www.hln.be/binnenland/ruim-6-miljoen-belgen-gebruiken-itsme-covid-safe-ticket-plaveide-weg-voor-identificatie-app~afb80ccac/) [base](https://www.hln.be/binnenland/zeven-miljoen-belgen-gebruiken-itsme~a825e29e/) once in a while, while all I hear from those users is that they were forced into using Itsme. To me it's very clear the added value isn't there for the user but in stead for the enterprises relying on Itsme. I'm afraid you have the answer to their we-dont-care-about-linux mindset there. In fact they don't care about you at all, no matter what system you use.

**Comment by sreedevk** [comment]
I can confirm that this works on the following setup for itsme activation with a Belgian ID:

OS: Arch linux
Browser: Mozilla Firefox 152.0
Connective signing extension 1.0.5
Reader: Alcor Link AK9563 (Thinkpad L14 Gen 6 builtin card reader)

Arch Linux Specific Details:
```
# necessary packages
sudo pacman -S --needed pcsclite ccid python-pyscard tk python-requests base-devel swig git

# Enable the smartcard daemon
sudo systemctl enable --now pcscd.socket

# check if the card is readable using
pcsc_scan
```

the others steps to build and install the browser extension are identical to what's described on the project's README.

To the owner of this project: 
Thank you so much for creating and maintaining this project. Without your work this simply wouldn't be possible on Linux, so it's genuinely appreciated.

**Comment by MarkRijckenberg** [comment]
Many thanks for this excellent solution! Very useful.

Itsme activation via your script, an eid card reader and a Belgian eid card worked very well on NixOS unstable yesterday.
I had to use Chromium/Google Chrome/Brave browser, because Firefox did not work with the Connective plugin in my case.

Platform details where it worked:

```
 OS -> NixOS
├ -> zokor 26.11 (Zokor)
├ -> Linux 7.2.2-cachyos-lto
└ -> bash 5.3.15

 DE/WM -> Muffin (X11)
├󰧨 -> lightdm 1.33.1 (X11)

󰌢 PC -> 20U3000NGE (ThinkPad L15 Gen 1)
├󰻠 -> Intel(R) Core(TM) i7-10510U (8) @ 4.90 GHz
├󰍛 -> Intel UHD Graphics @ 1.15 GHz [Integrated]

  Desktop: Cinnamon v: 6.6.9 Distro: NixOS 26.11 (Zokor)
  Device-1: Chicony Integrated Camera driver: uvcvideo type: USB
  Display: x11 server: X.org v: 1.21.1.24 driver: N/A resolution: N/A

Info:
  Info: Tools: api: eglinfo,glxinfo gpu: gputop, intel_gpu_top, lsgpu
  Info: quad core Intel Core i7-10510U [MT MCP] speed (MHz): avg: 4792
nix (Determinate Nix 3.22.2) 2.35.2

⋉ pcsc_scan 
PC/SC device scanner
V 1.7.5 (c) 2001-2026, Ludovic Rousseau <ludovic.rousseau@free.fr>
Using reader plug'n play mechanism
Scanning present readers...
0: Alcor Micro AU9540 00 00
                         
Mon Aug 31 20:11:09 2026
 Reader 0: Alcor Micro AU9540 00 00
  Event number: 0
```

Link to my integration of connective-plugin-linux inside my NixOS flake configuration:
https://gitlab.com/Rijckenberg/shell-scripts/-/tree/master/NixOS/packages/connective-plugin-linux?ref_type=heads
